### ISSUE TITLE
[Assistive tech and device support] Alert and status messages aren't announced without receiving focus. (11.22.1)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#27413]
#### VFS team name
Caregivers

#### VFS product name
Improve File Upload Error Messaging

#### Point of Contact/Reviewers
Josh Kim

---

### Platform Issue
Alert and status messages aren't announced without receiving focus.

### Issue Details
Not within scope to error messaging in particular, but significantly impacts upload experience. The upload progress bar does not announce to users unless they tab directly onto it. Even on focus, it does not provide good feedback to the user or any indication that a file has been uploaded.

### Supporting Guidance
[11.22](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Other References
WCAG SC 4.1.3_AA

---

### Platform Recommendation
The intervals should be announced to screen reader users via an aria-live region.

### VFS Team Tasks to Complete

- [ ] Comment on the ticket if there are questions or concerns
- [ ] VFS team closes the ticket when the issue has been resolved

**Comment by shiragoodman** [comment]
@joshkimux can you insert the .mov file? Thanks!

**Comment by joshkimux** [comment]
I ended up not recording a video for this one, I'll remove that off from here!

**Comment by dat-gcio** [comment]
@joshkimux While working on this, I noticed that the progress indicator only appears during the 'uploading' stage. It also only ever goes to 80%, and sits there until a success or error happens. Then it just goes away. Would you recommend just having the real statuses announced to the user? For example, we would announce 'Uploading your file' after the user selects a file. We would announce the error as it is now, if there is an error, and a 'File Uploaded' if it is a success.

**Comment by joshkimux** [comment]
@dat-gcio , great observation and question! 👍 

They should be able to hear not only the status and errors, but also the % uploaded. Using a live region may be a good option here with `aria-live="polite"`. Here's a [good example of an accessible bounded progress bar](https://labs.levelaccess.com/index.php/Progress_Bar_with_ARIA_Live_Regions) in use by Level Access.

**Comment by joshkimux** [comment]
@shiragoodman after working with @dat-gcio , we agreed that this would be best redirected to platform as a bug to be fixed as it is out of scope for this particular effort and would require more deliberate time and effort to fix 🙇  

I've removed the caregiver label and have added the components label-- is there any other steps I should take to redirect this as a platform component fix?

**Comment by shiragoodman** [comment]
thanks @joshkimux. I appreciate the follow through. No additional steps are needed at this time. I have created a [github ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/28061) for documenting this process so that we can ensure that the ticket is properly reassigned. Thanks again!

**Comment by shiragoodman** [comment]
@caw310 @k80bowman FYI this ticket somehow got overlooked. I added your team label. If you have questions, please let me know. Thanks!

**Comment by joshkimux** [comment]
I believe this was 10-10ez? @shiragoodman

**Comment by shiragoodman** [comment]
@joshkimux @caw310 this is the 1010-CG or Caregivers. We have added GH product label to all issue tickets so  in the future, check there first.

**Comment by caw310** [comment]
@SarahKay8 - this is a stretch goal for Sprint 5 (July 7 to 19)

**Comment by SarahKay8** [comment]
Hi @joshkimux can you provide a url at your earliest convenience?

**Comment by joshkimux** [comment]
@Saraharaba8 [log in instructions are included in the staging review ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/26697) underneath the "Artifacts" heading 😄

**Comment by SarahKay8** [comment]
Thanks! @joshkimux

**Comment by SarahKay8** [comment]
Hi @caw310
While focus is on the progress bar the current progress is still not being read out by screenreader.

**Comment by SarahKay8** [comment]
Hi all initial bug is still present. I would personally give a visually hidden `<span>` with aria-live (polite or assertive depends on the context) and aria-atomic + true, and have the text within the span and read "80% complete " or something like that. @caw310

**Comment by caw310** [comment]
Sarah and Katy discussed that there is a bigger issue with other components. There is a ticket to fix in sprint 6 so closing this.
